tests/Fixer/Casing/NativeConstTypeDeclarationCasingFixerTest.php

```diff
@@ -0,0 +1,197 @@
+                    const ?\INT\B E = null;
+                    public const ?INT\A/* x */X=C;
+                }
+            ',
+            '<?php
+                class Foo extends FooParent
+                {
+                    const ?OBJECT SOME_OBJECT = NULL;
+                    const ?PARENT SOME_PARENT = NULL;
+                    const ?Self SOME_SELF = NULL;
+                    const ?INT/* x */A/* y */= 3;
+
+                    const ?BAR B = null;
+                    const ?BAR C = null;
+                    const ?\BAR D = null;
+                    const ?\INT\B E = null;
+                    public const ?INT\A/* x */X=C;
+                }
+            ',
+        ];
+
+        yield 'simple `|`' => [
+            '<?php class Foo1 {
+                const D|float BAR = 1.0;
+            }',
+            '<?php class Foo1 {
+                const D|Float BAR = 1.0;
+            }',
+        ];
+
+        yield 'multiple `|`' => [
+            '<?php class Foo2 {
+                const int|array|null|A\B|\C\D|float BAR_1 = NULL;
+            }',
+            '<?php class Foo2 {
+                const INT|ARRAY|NULL|A\B|\C\D|FLOAT BAR_1 = NULL;
+            }',
+        ];
+
+        yield 'handle mix of `|` and `(X&Y)`' => [
+            '<?php
+                class Foo extends FooParent
+                {
+                    private const Z|A\S\T\R|int|float|iterable SOMETHING = 123;
+                    protected const \A\B|(Foo & Stringable )|null|\X D = null;
+
+                    public const Foo&Stringable G = V::A;
+                }
+            ',
+            '<?php
+                class Foo extends FooParent
+                {
+                    private const Z|A\S\T\R|INT|FLOAT|ITERABLE SOMETHING = 123;
+                    protected const \A\B|(Foo & Stringable )|NULL|\X D = null;
+
+                    public const Foo&Stringable G = V::A;
+                }
+            ',
+        ];
+
+        yield 'interface, nullable int' => [
+            '<?php interface SomeInterface {
+                const ?int FOO = 1;
+            }',
+            '<?php interface SomeInterface {
+                const ?INT FOO = 1;
+            }',
+        ];
+
+        yield 'trait, string' => [
+            '<?php trait T {
+                const string TEST = E::TEST;
+            }',
+            '<?php trait T {
+                const STRING TEST = E::TEST;
+            }',
+        ];
+
+        yield 'enum, int' => [
+            '<?php enum E {
+                const int TEST = 789;
+            }',
+            '<?php enum E {
+                const INT TEST = 789;
+            }',
+        ];
+
+        yield 'do not fix' => [
+            '<?php class Foo {
+                PUBLIC FUNCTION FOO_1(INT|FLOAT|NULL|ITERABLE $A): VOID {}
+
+                PUBLIC CONST FOO&STRINGABLE G = A::B;
+            }
+
+            enum E {
+                PUBLIC CONST STATIC A = E::FOO;
```

Comment:
`enum`'s are now really on another level in PHP :sweat_smile:

Reply:
what static is doing here? it's not about marking constant to be static, right? it's more to inherit the type from E::FOO ?

Reply:
As far as I remember enums can't be extended, so `static` is not possible (or at least does not make sense) there.

Reply:
and you know what static keyword is doing for this constant?

Reply:
I believe it's type definition for the const (since this fixer is about it), but since enum can't be extended it just can't represent child class. It's invalid scenario.

Reply:
yeah `static`  would be the type hint here, yet it makes no sense, but for now it is also valid syntax.. maybe this will be disallowed when PHP8.3 gets released

Reply:
For me [it looks invalid](https://twitter.com/_Codito_/status/1680878870189297665):
- enums are not extendable
- `static` was added as a available _return_ type, it can't be used as type elsewhere

Reply:
https://wiki.php.net/rfc/typed_class_constants
```
enum E {
    // This is legal since constants provide a covariant context
    public const static A = E::Foo;
```

:shrug:

Reply:
But still enums can't be extended, so `static` does not make sense 🙂. Can we drop this now and see how it goes in 8.3 development? I believe we should have only valid use cases covered and this one looks fishy to me.

Reply:
it is explicit part of the RFC and this case passes on PHP8.3 builds as valid syntax, why take it out?

Reply:
Because [it does not make sense](https://onlinephp.io/c/db35f):

<img width="1244" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/6b0a6e24-4cb1-496b-9f9b-5e2bc7b5dced">

PHP 8.3 is still under development, it's only on `alpha` stage, so it can contain bugs and invalid assumptions. Personally I would wait for core team's answer to [this](https://github.com/php/php-src/pull/10444#issuecomment-1637959544).

Reply:
BTW. "_it is explicit part of the RFC and this case passes on PHP8.3 builds as valid syntax_" - there is generic type used in [this RFC](https://wiki.php.net/rfc/local_variable_types), does it mean we should prepare for it 😉? Yeah, this one is in draft, while the former is accepted, but still 🤷‍♂️. I would rather treat RFC's as entry point for covering changes, but wouldn't take everything for granted, especially for unreleased versions.

Reply:
I talked about a merged change, not a draft and about documentation with that change that explicit gives this as an example.
Lets wait for the answer and PHP8.3 being released, there is rush.